app/controllers/users/registrations_controller.rb

```diff
@@ -9,5 +16,15 @@ def update
         respond_with resource
       end
     end
+
+    private
+
+    def their_own?
+      current_user.id.eql?(params[:id].to_i)
+    end
+
+    def authorize_for_current_user
+      authorize current_user if their_own?
```

Comment:
Yep, but it's the policy who has to decide that. So the `UserPolicy` should have something like this:

```
def edit_password?
  user_themselves?
end
```

And then you're authorising a `user` (which comes from the params) in context of the `current_user`, which then only returns `true` when it's the same user.